File: java/test/com/github/bazel_contrib/contrib_rules_jvm/javaparser/generators/workspace/com/gazelle/java/javaparser/generators/BareClassMethodReceiver.java
package workspace.com.gazelle.java.javaparser.generators;

// No import — SamePackageHelper is a same-package class used as a bare method receiver.
// visitMethodInvocation only handles MemberSelectTree receivers (a.B.method()),
// not IdentifierTree receivers (B.method()). So SamePackageHelper.create()
// is invisible to the method-invocation detection path, and without an import,
// the same-package fallback in checkFullyQualifiedType is never reached.

public class BareClassMethodReceiver {
    public void doWork() {
        SamePackageHelper.create();
    }
}

File: java/test/com/github/bazel_contrib/contrib_rules_jvm/javaparser/generators/workspace/com/gazelle/kotlin/javaparser/generators/BareClassMethodReceiver.kt
package workspace.com.gazelle.kotlin.javaparser.generators

// No import — SamePackageHelper is in the same package but may be in a different
// Bazel target (split package). The KtParser relies entirely on imports for
// usedTypes; visitSimpleNameExpression sees "SamePackageHelper" and recognises
// it as a likely class name, but never adds it to usedTypes. Without a
// same-package fallback (like ClasspathParser's checkFullyQualifiedType),
// this dependency is invisible.

class BareClassMethodReceiver {
    fun doWork() {
        SamePackageHelper.create()
    }
}
